lib/main.rb

```diff
@@ -0,0 +1,17 @@
+def bubble_sort(numbers)
+  sorted = true
+
+  (numbers.length - 1).times do |i|
+    sub_arr = [numbers[i], numbers[i + 1]]
+
+    next if sub_arr[0] <= sub_arr[1]
+
+    numbers = numbers[0, i] + sub_arr.sort! + numbers[i + 2...]
+    sorted = false
+    break
```

Comment:
I think this `break` shouldn't be here. See on the gif in the readme, that the algorithm compares each pair of numbers up from the beginning until the end of the array, and then it starts again on the beginning. Here, this `break` statement is not letting the comparison get to the end of the array; instead, it gets only to the first pair that requires a position swap, then the iteration breaks and the recursive call makes the iteration start over from the beginning of the array

Reply:
I agree. With that "break", I'll be restarting the break each we sort a pair of numbers. But I need to go through all the array's indexes, sorting each pair of numbers

![image](https://user-images.githubusercontent.com/26731448/188990714-80c50e74-17d2-44bd-b219-10e6ba4194fc.png)